plasmapy/diagnostics/tests/test_langmuir.py

```diff
@@ -1,40 +1,181 @@
-"""Tests for functions that use Langmuir analysis functions."""
+# coding=utf-8
+"""Tests for Langmuir probe analysis functions."""
 
 import numpy as np
+from astropy import units as u
+import pytest
+import plasmapy.constants as const
 
-from plasmapy.diagnostics.langmuir import (swept_probe_analysis,
-                                           obtain_EEDF,
-                                           obtain_EEPF)
-
-
-class Test_swept_probe_analysis():
-    def test_no_signal(self):
-        """
-        Checks whether an error is raised when zero signal is passed to the
-        function.
-        """
-        errStr = ("Performing a swept probe analysis should return an error "
-                  "when zero signal is passed, but doesn't.")
-        assert True, errStr
-
-
-class Test_obtain_EEDF():
-    def test_no_signal(self):
-        """
-        Checks whether an error is raised when zero signal is passed to the
-        function.
-        """
-        errStr = ("Attempting to obtain the EEDF should return an error when "
-                  "zero signal is passed, but doesn't.")
-        assert True, errStr
-
-
-class Test_obtain_EEPF():
-    def test_no_signal(self):
-        """
-        Checks whether an error is raised when zero signal is passed to the
-        function.
-        """
-        errStr = ("Attempting to obtain the EEPF should return an error when "
-                  "zero signal is passed, but doesn't.")
-        assert True, errStr
+from plasmapy.diagnostics.langmuir import (Characteristic,
+                                           swept_probe_analysis)
+
+N = 30  # array length of dummy probe characteristic
+
+current_arr = np.random.rand(N) * u.A
+bias_arr = np.random.rand(N) * u.V
+
+
+class Test__characteristic_errors:
+    r"""Test the Characteristic class constructor in langmuir.py"""
+
+    bias_infarr = np.append(np.random.rand(N - 1), np.inf) * u.V
+    current_infarr = np.append(np.random.rand(N - 1), np.inf) * u.A
+
+    bias_longarr = np.random.rand(N + 1) * u.V
+    current_longarr = np.random.rand(N + 1) * u.A
+
+    current_arr2 = np.random.rand(N) * u.A
+
+    def test_infinite_I(self):
+        r"""Test error upon an infinite current value"""
+
+        with pytest.raises(ValueError):
+            Characteristic(bias_arr, self.current_infarr)
+
+    def test_infinite_U(self):
+        r"""Test error upon an infinite bias value"""
+
+        with pytest.raises(ValueError):
+            Characteristic(self.bias_infarr, current_arr)
+
+    def test_unequal_arrays(self):
```

Comment:
This seems to be two tests in one. If the first one fails (or doesn't fail in this case) then it won't catch the second one. Are these two tests dependent on each other or can than be sensibly placed into separate methods?

Reply:
:fireworks:  :fireworks:  :us: :us: :us: :fireworks: :fireworks: 
**!!!TEST INDEPENDENCE EXTRAVAGANZA!!!**
:fireworks:  :fireworks:  :us: :us: :us: :fireworks: :fireworks: